### PR TITLE
Add back inductor

### PR DESCRIPTION
**Pull request by padauker**
#1168

### CODE REVIEW

**Review by TGCrystal — DISMISSED** [review]

**Review by TGCrystal — APPROVED** [review]

File: src/app/analog/views/index.ts [hunk 1]
```diff
@@ -32,6 +32,9 @@ export const Views: ViewRecord<AnalogObj, AnalogViewInfo> = {
 
     "Ground":   (c, o) => new ComponentView(c, o, V(1.2, 0.6), "ground.svg"),
     "Resistor": (c, o) => new ComponentView(c, o, V(1.2, 1), "resistor.svg"),
+    "Inductor": (c, o) => new ComponentView(c, o, V(0.8, 240*0.8/104), "inductor.svg"), 
+    "CurrentSource": (c, o) => new ComponentView(c, o, V(1, 1), "currentsource.svg"),
+
```
Comment (LeonMontealegre) [inline src/app/analog/views/index.ts:37]: Keep these vertically aligned at `(c, o)`

File: src/app/core/models/types/analog.ts [hunk 2]
```diff
@@ -24,6 +32,10 @@ export const DefaultAnalogComponent: { [C in AnalogComponent as C["kind"]]: Comp
     "AnalogNode": (id) => ({ ...DefaultComponent(id), kind: "AnalogNode"                 }),
     "Ground":     (id) => ({ ...DefaultComponent(id), kind: "Ground"                     }),
     "Resistor":   (id) => ({ ...DefaultComponent(id), kind: "Resistor", resistance: 1000 }),
+    "Inductor":   (id) => ({ ...DefaultComponent(id), kind: "Inductor", inductance: 10 }),
+    "CurrentSource": (id) => ({ ...DefaultComponent(id), kind: "CurrentSource", amperage: 1 }),
```
Comment (LeonMontealegre) [inline src/app/core/models/types/analog.ts:36]: keep these all vertically aligned at
`(id) =>`
and similarly, keep the end of the lines vertically aligned with the `}),`

File: src/app/core/views/portinfo/analog/index.ts [hunk 3]
```diff
@@ -36,4 +36,22 @@ export const AnalogPortInfo: PortInfoRecord<AnalogComponent> = {
             ],
         }],
     },
+    "Inductor": {
+        ...DefaultAnalogPortInfo,
+        PositionConfigs: [{
+            "ports": [
+                { origin: V(0, -0.925), target: V(0, -1.425), dir: V(0, -1) },
+                { origin: V(0, +0.925), target: V(0, +1.425), dir: V(0, +1) },
```
Comment (LeonMontealegre) [inline src/app/core/views/portinfo/analog/index.ts:44]: these values should probably be more explicit as they depend on the height of the inductor which we know is `240*0.8/104` (which comes from the size of the `inductor.svg` image btw), so like the origin y should be like `240/104*0.8/2`, you can (and probably should) even just store a constant at the top of this file called `INDUCTOR_HEIGHT` which is set to `240/104*0.8`

File: src/app/core/views/portinfo/analog/index.ts [hunk 4]
```diff
@@ -36,4 +36,22 @@ export const AnalogPortInfo: PortInfoRecord<AnalogComponent> = {
...
+        }],
+    },
+    "CurrentSource": {
+        ...DefaultAnalogPortInfo,
+        PositionConfigs: [{
+            "ports": [
+                { origin: V(0, -0.5), target: V(0, -1.3), dir: V(0, -1) },
+                { origin: V(0, +0.5), target: V(0, +1.3), dir: V(0, +1) },
+            ],
+        }],
```
Comment (LeonMontealegre) [inline src/app/core/views/portinfo/analog/index.ts:55]: If you want to go above and beyond here, you could create a utility method in `app/core/views/portinfo/positioning/utils.ts` called like `TwoPortPositioner` which would be like:
```ts
export function TwoPortPositioner(dir: "horizontal" | "vertical", size: number): PortConfig[]
```
which would be used like
```ts
"Resistor": {
    ...
    PositionConfigs: [{ "ports": TwoPortPositioner("horizontal", 1.2) }],
}
...
"Inductor": {
    ...
    PositionConfigs: [{ "ports": TwoPortPositioner("vertical", 240/104*0.8) }],     
},
...
"CurrentSource": {
    ...
    PositionConfigs: [{ "ports": TwoPortPositioner("vertical", 1) }],
},
...
```

**Review by LeonMontealegre — CHANGES_REQUESTED** [review]
Looking good! Just a few nits and recommendations!

**Pull request closed without merging**